src/dll.py

```diff
@@ -40,23 +40,33 @@ def append(self, data):
         new_node._next = None
         new_node._prev = self.tail
         if self.tail:
-            self.tail._next = new_node._prev
+            self.tail._next = new_node
         self.tail = new_node
         if self.head is None:
             self.head = new_node
 
     def pop(self):
         """Remove data from the head of the list and return it."""
         pop_node = self.head
```

Comment:
It's not made explicit in the assignment, but `pop` and `shift` should raise the appropriate error when trying to pop an empty DLL.